### PR TITLE
Resolve conditionals with active job

### PR DESCRIPTION
**Pull request by custode**
Depends on #894 

Schedule the conditional resolving service with active job instead of computing them directly.

### CODE REVIEW

File: lib/extensions/conditional/active_record/base.rb [hunk 1]
```diff
@@ -62,7 +62,11 @@ def dependent_class
       raise NotImplementedError
     end
 
-    def resolve_conditional_for(_course_user)
+    # Resolve the conditionals for the given course user
+    #
+    # @param [CourseUser] course_user The user with conditionals to be resolved
```
Comment (lowjoel) [inline lib/extensions/conditional/active_record/base.rb]: Should this document the fact that the job is returned, and that people can wait for the job to finish?

**Comment by lowjoel** [comment]
@custode I think we need a rebase

**Comment by custode** [comment]
@lowjoel Rebased and added the eager loading.

**Comment by lowjoel** [comment]
@custode sorry, I didn't think about it deep enough. Is ConditionalResolvingService better or ConditionalResolutionService? I think ResolutionService sounds more grammatically correct (sorry, nitpicking here)

File: lib/extensions/conditional/active_record/base.rb [hunk 2]
```diff
@@ -62,11 +62,12 @@ def dependent_class
       raise NotImplementedError
     end
 
-    # Resolve the conditionals for the given course user
+    # Resolve and update the conditionals for the given course user.
     #
     # @param [CourseUser] course_user The user with conditionals to be resolved
+    # @return [Course::Conditional::ConditionalResolvingService] The resolver's job instance
```
Comment (lowjoel) [inline lib/extensions/conditional/active_record/base.rb]: I think this is the wrong type it returns the job, not the service object.

Comment (custode) [inline lib/extensions/conditional/active_record/base.rb]: My bad. Thanks!

**Comment by lowjoel** [comment]
@custode with this, your job is done? Or are we blocking on APIs?

**Pull request merged**